### PR TITLE
Add a `functional` test marker

### PR DESCRIPTION
**Pull request by whit537**
Part of #903. ~~Includes #1007.~~

### CODE REVIEW

**Comment by whit537** [comment]
@kapilt I marked all of the tests in vpc except for [one](https://github.com/capitalone/cloud-custodian/blob/67790ba5fa51717569e0c8baed81366b2185d74d/tests/test_vpc.py#L522-L535) that had a comment pretty clearly indicating pre-existing context. Figured you'd have an easier/faster time than I identifying others that may in fact not be candidates here.

**Comment by kapilt** [comment]
alot of the markers here are suspect, nutshell if a unit test isn't creating the resource being evaluated in a policy, ideally without pre-conditions, then its not a good fit functional. nutshell we should probably just do this as functional marker, plus vpc tests that have a create vpc in their method body, to start.

**Comment by whit537** [comment]
Rebased now that #1007 is in, was 67790ba5fa51717569e0c8baed81366b2185d74d.

File: tests/test_vpc.py [hunk 1]
```diff
@@ -775,6 +795,7 @@ def test_cidr_size_egress(self):
               u'ToPort': 443,
               u'UserIdGroupPairs': []}])
 
+    @functional
```
Comment (kapilt) [inline tests/test_vpc.py]: this isn't really a functional test, it doesn't interact with the api.

File: tests/test_vpc.py [hunk 2]
```diff
@@ -592,6 +608,7 @@ def test_multi_attribute_ingress(self):
         manager = p.get_resource_manager()
         self.assertEqual(len(manager.filter_resources(resources)), 1)
 
+    @functional
```
Comment (kapilt) [inline tests/test_vpc.py]: this test is not a good functional candidate, its has pre-condition dependencies.

File: tests/test_vpc.py [hunk 3]
```diff
@@ -563,6 +578,7 @@ def test_only_ports_ingress(self):
...
```
Comment (kapilt) [inline tests/test_vpc.py]: not a good functional candidate, per previous pr comments.

**Comment by kapilt** [comment]
the decorator looks good, but the use of the decorator is not so good, please remove the functional markers in vpc or use the guidelines in previous comments on this pr when decorating, i checked 5 decorated tests, and found 4 bad ones.

**Comment by whit537** [comment]
Getting back to this ...

**Comment by whit537** [comment]
> nutshell we should probably just do this as functional marker, plus vpc tests that have a create vpc in their method body, to start.

Done in c7bfff196b93385fde935e6ddb79bbef2634940e.

**Comment by whit537** [comment]
Rebased, was c7bfff196b93385fde935e6ddb79bbef2634940e.

**Pull request merged**